File: examples/src/main/javascript/nodejs/events/EventEmitting.js
/**
 * Event Emitting
 * @author lawrence.daniels@gmail.com
 */
(function() {
    const EventEmitter = require("events");
    const ee = new EventEmitter();

    setImmediate(function () {
        // This will crash the process because no "error" event
        // handler has been added.
        ee.emit("error", new Error("This will crash"));
    });
    
})();

File: examples/src/main/scala/examples/nodejs/events/EventEmitting.scala
package examples.nodejs.events

import com.github.ldaniels528.meansjs.nodejs._
import com.github.ldaniels528.meansjs.nodejs.events.EventEmitter

/**
  * Event Emitting Example
  * @author lawrence.daniels@gmail.com
  */
class EventEmitting(require: Require) {
  val eventEmitter = require[EventEmitter]("events")
  val ee = eventEmitter()

  setImmediate(() => {
    // This will crash the process because no "error" event
    // handler has been added.
    ee.emit("error", Error("This will crash"))
  })

}
